common/txmgr/broadcaster.go

```diff
@@ -792,13 +797,26 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) save
 }
 
 // Used to get the next usable sequence for a transaction
-func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetNextSequence(address ADDR) (seq SEQ, err error) {
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetNextSequence(ctx context.Context, address ADDR) (seq SEQ, err error) {
 	eb.sequenceLock.Lock()
 	defer eb.sequenceLock.Unlock()
 	// Get next sequence from map
 	seq, exists := eb.nextSequenceMap[address]
 	if !exists {
-		return seq, errors.New(fmt.Sprint("address not found in next sequence map: ", address))
+		eb.logger.Criticalw("address not found in local next sequence map. Attempting to search and populate sequence.", "address", address.String())
+		// Check if address is in the enabled address list
+		if slices.Contains(eb.enabledAddresses, address) {
```

Comment:
If an address is not part of `enabledAddresses`, then we return an oqaue error? But we don't even check that if it was in `nextSequenceMap` already? That seems inconsistent.

Why don't we invert this check, and return a clear error for that case?
```suggestion
		if !slices.Contains(eb.enabledAddresses, address) {
			return seq, fmt.Errorf("address disabled: %s", address)
		}
```
Then afterwards, if we still find `foundSeq == nil`, then we have a real failure error.

Reply:
Inverting the `!exists` check would simplify things too:
```go
	if exists {
		return seq, nil
	}
	eb.logger.Infow("address not found in local next sequence map. Attempting to search and populate sequence.", "address", address.String())
	if !slices.Contains(eb.enabledAddresses, address) {
		return seq, fmt.Errorf("address disabled: %s", address)
	}
```

Reply:
The reason for not checking the enabled address list before checking the map is because the map was populated using the enabled address list on initialization. So if we find an address already in the map we know it's valid. But for sanity I could move the check ahead of it if you'd like. Otherwise I've addressed the other suggestions here